manticore/native/models.py

```diff
@@ -140,3 +142,123 @@ def strlen(state, s):
             ret = ITEBV(cpu.address_bit_size, byt == 0, offset, ret)
 
     return ret
+
+
+def is_NULL(byte, constrs) -> bool:
+    """
+    Checks if a given byte read from memory is NULL or effectively NULL
+
+    :param byte: byte read from memory to be examined
+    :param constrs: state constraints
+    :return: whether a given byte is NULL or constrained to NULL
+    """
+    if issymbolic(byte):
+        return not Z3Solver.instance().can_be_true(constrs, byte != 0)
```

Comment:
Note, you might be able to rephrase this using `Z3Solver.instance().must_be_true`.  But I haven't used that API, so I can't comment further on performance implications, behavioral differences, etc :)

Reply:
Taking a look at this. I'm also not super well versed with the API.

Reply:
Yes that cut the python test time in half! Good catch!